[PT FE]: Add support for prim::data operation

Sujanian1304 opened the pull request:
this solves issue #29721 
### **Details:**
 -added support for prim::data
### Tickets:
 - Closes [Good First Issue]: #29721

mvafin commented on src/frontends/pytorch/src/op_table.cpp:
```diff
@@ -795,6 +795,7 @@ const std::unordered_map<std::string, CreatorFunction> get_supported_ops_ts() {
         {"prim::TupleIndex", op::translate_tuple_index},
         // prim::TupleUnpack - Supported in limited set of patterns
         {"prim::type", op::skip_node},  // Used with prim::device, pass PtFrameworkNode.
+        {"prim::data", op::translate_data},
```
I think it can be `op::skip_node`. Doesn't think you need any special handling for complex types in this case.

mvafin commented on tests/layer_tests/pytorch_tests/test_data.py:
```diff
@@ -0,0 +1,45 @@
+        tensor = torch.from_numpy(data).to(self.dtype)
+        return (tensor.numpy(),)
+
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.float64, torch.int32, torch.int64])
```
Add test case for complex numbers too

Sujanian1304 commented:
hey @mvafin ,thanks for the feedback.I have addressed all of your suggestions and added test cases for complex-types,changed to op::skip_node.Please let me know if any other changes are needed.
